### PR TITLE
MKB-553: documents search

### PR DESCRIPTION
**Pull request by radoslavzeman**
(commit by commit)

Refactor search in "Documents and disclosure" page to use meilisearch instead of grapghql filtering.
- add new meili fetcher, add more filterable and sortable attributes
- copy some components from Marianum (select, sorting) - this may need some revisit later
- the whole UX is simplified
  - there in no file-category page - filtering by category is now on main Documents and disclosure page
  - category buttons are replaced by select field
  - some filed were removed and renamed
    - `/pages/file/[slug]` which represented page for Basic Document detail is now `/pages/basic-document/[slug]`
    - `/pages/documents/[slug]` which represented file-category pages is now removed
- also URLs are siplified
  - from: `/o-nas/dokumenty-a-zverejnovanie-informacii/:categorySlug/:slug`
  - to: `/o-nas/dokumenty-a-zverejnovanie-informacii/:slug`
  - this also solves the bug with `undefined` in urls (as categorySlug)
  - english page slug is now used on EN local instead of slovak slug
    - from: `/en/o-nas/dokumenty-a-zverejnovanie-informacii/...`
    - to: `/en/about-us/documents-and-public-disclosure-of-information/...`
  - redirects are added to next config - will be changed to permanent when tested properly

### CODE REVIEW

File: next/components/Atoms/SelectField.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import ChevronDown from '@assets/images/chevron-down.svg'
+import FieldWrapper from '@components/Molecules/FieldWrapper'
+import { Listbox } from '@headlessui/react'
+import { isDefined } from '@utils/isDefined'
+import cx from 'classnames'
+import { ReactNode, useCallback, useId, useMemo, useState } from 'react'
+import { usePopper } from 'react-popper'
```
Comment (mpinter) [inline next/components/Atoms/SelectField.tsx:7]: good to know about this

File: next/components/Atoms/SelectField.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import ChevronDown from '@assets/images/chevron-down.svg'
+import FieldWrapper from '@components/Molecules/FieldWrapper'
+import { Listbox } from '@headlessui/react'
```
Comment (mpinter) [inline next/components/Atoms/SelectField.tsx:3]: I see headlessui being used - not a problem with that, just wondering what is our consensus on these libs. Is it "react-aria unless it does not have the right stuff, then pick from others, then code your own" ?

File: next/components/pages/DocumentsPage.tsx [hunk 3]
```diff
@@ -1,214 +1,129 @@
...
-      if (event.key === 'Enter') {
-        handleSearchReset()
-      }
-    },
-    [handleSearchReset]
-  )
+  useEffect(() => {
+    setFilters({ ...filters, searchValue })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchValue])
```
Comment (mpinter) [inline next/components/pages/DocumentsPage.tsx:62]: might be nicer to use the calback version of setState and avoid the eslint-ignore:

```suggestion
useEffect(() => {
    setFilters((prevFilters) => ({ ...prevFilters, searchValue }))
  }, [searchValue])
```

File: next/next.config.js [hunk 4]
```diff
@@ -62,6 +62,19 @@ const nextConfig = {
...
+        permanent: false,
+        locale: false,
+      },
+      {
+        source: '/en/o-nas/dokumenty-a-zverejnovanie-informacii/:categorySlug/:slug',
+        destination: '/en/about-us/documents-and-public-disclosure-of-information/:slug',
+        permanent: false,
+        locale: false,
+      },
+
```
Comment (mpinter) [inline next/next.config.js:77]: I think you can mark both as permanent

```suggestion
      {
        source: '/o-nas/dokumenty-a-zverejnovanie-informacii/:categorySlug/:slug',
        destination: '/o-nas/dokumenty-a-zverejnovanie-informacii/:slug',
        permanent: true,
        locale: false,
      },
      {
        source: '/en/o-nas/dokumenty-a-zverejnovanie-informacii/:categorySlug/:slug',
        destination: '/en/about-us/documents-and-public-disclosure-of-information/:slug',
        permanent: true,
        locale: false,
      },

```

File: next/next.config.js [hunk 5]
```diff
@@ -62,6 +62,19 @@ const nextConfig = {
   },
   async redirects() {
     return [
+      {
+        source: '/o-nas/dokumenty-a-zverejnovanie-informacii/:categorySlug/:slug',
+        destination: '/o-nas/dokumenty-a-zverejnovanie-informacii/:slug',
```
Comment (mpinter) [inline next/next.config.js:67]: this redirect did not seem to work for me for some reason ?

https://user-images.githubusercontent.com/2327312/209363060-5a845286-46ea-43a0-8a5f-5a26c0178960.mov

was running this against staging instance of strapi and meilisearch (copied the whole staging env as my .env.development.local), so just wanted to recheck if that is the reason why I'm still getting the `undefined`s in the result

**Review by mpinter — APPROVED** [review]
Please check the last comment (on the search result urls and redirect), that one may need resolution before this is deployed - but otherwise looking good, gj and good to go 🚢

File: next/components/Atoms/SelectField.tsx (hunk 2 above)
Comment (radoslavzeman) [reply next/components/Atoms/SelectField.tsx:3]: This component was just copied from Marianum. We plan to refactor it to use react-aria, but it was not task for this PR. I know it is not ideal to introduce "more work" into the project, but as the refactoring will be the same for both project, I think it is reasonable quick solution to use this component.

File: next/components/pages/DocumentsPage.tsx (hunk 3 above)
Comment (radoslavzeman) [reply next/components/pages/DocumentsPage.tsx:62]: Yes, for sure :) thx

File: next/next.config.js (hunk 4 above)
Comment (radoslavzeman) [reply next/next.config.js:77]: Okay

File: next/next.config.js (hunk 5 above)
Comment (radoslavzeman) [reply next/next.config.js:67]: Thanks, there was a mistake in redirects due to disabled locale detection. Adding '/sk/` prefix to source url fixed it.
You also found another bug with undefined category in reach results - fixed and removed too.

**Pull request merged**